README.md

```diff
@@ -984,6 +984,10 @@ in inheritance.
   is a hybrid of `Array`'s intuitive inter-operation facilities and
   `Hash`'s fast lookup.
 
+*  Prefer `map` over `collect`, `detect` over `find`, and `reduce` over
```

Comment:
I like @tommydangerous's bulleted version a bit more. Seems more digestible.

Overall, I think these are good suggestions.

Re: @kmsun07's question about whether the first 3 are arbitrary (and no, I don't have numbers to back these up; it's just based on what I've seen at different companies and open source projects):

* I'm pretty sure the Ruby community uses `map` far more often than `collect`, so it's in keeping with community conventions (and that's one of the goals of this style guide).
* I think the Ruby community uses `find` slightly more than `detect`, but I am in favor of preferring `detect`, as suggested in this PR, to make it clear it's not an ActiveRecord call. That's an actual practical reason as opposed to just personal preference.
* I think the Ruby community uses `inject` slightly more than `reduce`, but you know, there's map/reduce and all that, so it seems kind of funky to use `inject`. Besides, `reduce`/`inject` is not used nearly as often as `map`/`collect`.